### PR TITLE
New pull Request

### PR DESCRIPTION
**Pull request by geethapalyam**
body

### CODE REVIEW

File: index-refactor.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        server.ehlo() # Can be omitted
+        server.starttls(context=context) # Secure the connection
+        server.ehlo() # Can be omitted
+        server.login(sender_email, password)
+        server.sendmail(sender_email, receiver_email, message)
+    except Exception as e:
+        # Print any error messages to stdout
+        print(e)
+    finally:
+        server.quit()
```
Comment (NoamGoren) [inline index-refactor.py:94]: can be refactored to the following 

    smtp = smtplib.SMTP_SSL(host="smtp.sendgrid.net", port=465)
    sendgrid_user, sendgrid_pw  = get_email_credentials()
    smtp.login(user=sendgrid_user, password=sendgrid_pw)
    smtp.send_message(msg, msg['From'], email)
    smtp.quit()


also, do we need to keep  server.ehlo() # Can be omitted?

File: index-refactor.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        tree = repo.create_git_tree([element1], base_tree)
+        parent = repo.get_git_commit(sha=branch_sha)
+
+        print("commiting the changes to the git repo")
+        commit = repo.create_git_commit("Version and checksum updated", tree, [parent])
+        branch_refs = repo.get_git_ref("heads/"+branch_name)
+        branch_refs.edit(sha=commit.sha)
+
+        print("creating the PR to the changes")
+        pr = repo.create_pull(title="New pull Request", body="body", head=branch_name, base="main")
```
Comment (NoamGoren) [inline index-refactor.py:130]: can be moved to a function

**Review by NoamGoren — CHANGES_REQUESTED** [review]

File: index-refactor.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+receiver_email = "palyamgeetha@gmail.com"
+#branch name
+branch_name = "runner"
+#slack channel receives notification
+channel_name = "#git-script"
+
+def sendSlackNotification(pr_number):
+    # Code to send alert to the slack channel
+    url = webhook_url
+    message = ("new Pull Request Link: https://github.com/{}/{}/pull/{}".format(username, repoName, pr_number))
```
[the remark this answers is not in the document]
Comment (geethapalyam) [reply index-refactor.py:42]: Sure, good idea.

File: index.py [hunk 4]
```diff
@@ -5,34 +5,44 @@
 import requests
```
[the remark this answers is not in the document]
Comment (geethapalyam) [reply index.py:5]: No, I will exclude this file while cloning to the adsk git

File: index-refactor.py (hunk 2 above)
Comment (geethapalyam) [reply index-refactor.py:130]: Sure, let me check.

File: index-refactor.py (hunk 1 above)
Comment (geethapalyam) [reply index-refactor.py:94]: ok

**Pull request closed without merging**